### ISSUE TITLE
twisted.words.test.test_irc.ClientTests.test_noticedDoesntPrivmsg is incorrect

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5330]
|<img alt="Wolf's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Wolf reported|
|-|-|
|Trac ID|trac#5330|
|Type|defect|
|Created|2011-10-22 11:12:09Z|

test_noticedDoesntPrivmsg calls ```irc_NOTICE('spam', "I don't want any spam!")```, but the code in irc_NOTICE is:
```
    def irc_NOTICE(self, prefix, params):
        """
        Called when a user gets a notice.
        """
        user = prefix
        channel = params[0]
        message = params[-1]
```

so, unless I'm missing something, the result is user == 'spam', channel == 'I', message == '!'.

I noticed this because on Python 3 params[-1] is 33, and that breaks later with message[0].

Passing a list with [chan, message] solves the problem.

Patch attached.

Attachments:

* [issue4419.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/2b9/2b9b2632a4a8177659a454b4fe0c448afb1ce359/8519e6a3f1bcc3fe97ba41d9367a62b3d1636845.diff) (552 bytes) - added by Wolf on 2011-10-22 11:12:26Z - Patch to fix the test.

<details><summary>Searchable metadata</summary>

```
trac-id__5330 5330
type__defect defect
reporter__Wolf Wolf
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__irc_noticed irc noticed
time__1319281929000000 1319281929000000
changetime__1320243175000000 1320243175000000
version__None None
owner__exarkun exarkun
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks!  This looks like a good fix to me.  I'll apply this.  Also, I'll just note that this mistake was introduced in r29441 for ticket [#4419](https://github.com/twisted/twisted/issues/4419).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [33050]) Fix (mostly harmless) incorrect argument used in NOTICE handling test

Author: Wolf
Reviewer: exarkun
Fixes: [#5330](https://github.com/twisted/twisted/issues/5330)

Correct a mistaken test which previously passed in a string where a
list was expected.
